addons/l10n_cl/models/account_move.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.exceptions import ValidationError
+
+from odoo import models, fields, api, _
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_latam_document_type_id_code = fields.Char(related='l10n_latam_document_type_id.code', string='Doc Type')
+    partner_id_vat = fields.Char(related='partner_id.vat', string='VAT No')
+    l10n_latam_internal_type = fields.Selection(related='l10n_latam_document_type_id.internal_type',
+                                                string='L10n Latam Internal Type')
+
+    def get_document_type_sequence(self):
+        """ Return the match sequences for the given journal and invoice """
+        self.ensure_one()
+        if self.journal_id.l10n_latam_use_documents and self.l10n_latam_country_code == 'CL':
+            res = self.journal_id.l10n_cl_sequence_ids.filtered(
+                lambda x: x.l10n_latam_document_type_id == self.l10n_latam_document_type_id)
+            return res
+        return super().get_document_type_sequence()
+
+    def _get_l10n_latam_documents_domain(self):
+        self.ensure_one()
+        domain = super()._get_l10n_latam_documents_domain()
+        if (self.journal_id.l10n_latam_use_documents and
+                self.journal_id.company_id.country_id == self.env.ref(
+                    'base.cl')):
+
+            journal_sequence_ids =self.env['ir.sequence'].search([
+                ('id', 'in', self.journal_id.l10n_cl_sequence_ids.ids)
+            ]).mapped('l10n_latam_document_type_id').ids
+
+            domain += [
+                ('active', '=', True),
+                ('id', 'in', journal_sequence_ids)
+            ]
+            if self.partner_id.l10n_cl_sii_taxpayer_type == '3':
+                domain += [('code', 'in', ['35', '38', '39', '41'])]
+        return domain
+
+    @api.constrains('type', 'l10n_latam_document_type_id')
+    def _check_invoice_type_document_type(self):
+        super()._check_invoice_type_document_type()
+        for rec in self.filtered('l10n_latam_document_type_id'):
+            tax_payer_type = rec.partner_id.l10n_cl_sii_taxpayer_type
+            latam_document_type_code = rec.l10n_latam_document_type_id.code
+            if not tax_payer_type and latam_document_type_code not in {
+                '35', '38', '39', '41'
```

Comment:
'not in' with a dict?  Why not a list?

Reply:
It's not a dict, It's a set -> set(). It's faster than list for in / not in searches

Reply:
55b3d5d9661043cb5445a7b63ab915981e0f7554

Reply:
[FIX] again, list to set (sorry) commit 6c2c96